### ISSUE TITLE
Trailing slash breaks workspace features when yarn run inside workspace folder

### ISSUE THREAD

**Issue by Y0hy0h** [yarnpkg/yarn#5816]
**Do you want to request a *feature* or report a *bug*?**

Bug.

**What is the current behavior?**
When the `workspaces` definition contains a trailing slash, running e. g. `yarn workspaces info` in the root works just as expected. But when you `cd` into the workspace folder and run `yarn workspaces info`, it fails because it cannot detect that that folder is a workspace.

**If the current behavior is a bug, please provide the steps to reproduce.**

1. Create a new project directory.
2. Create a subdirectory called `package`.
3. `yarn init` inside `package/`.
4. `yarn init` in the root.
5. Add `package/` (with trailing slash) as a workspace to the root `package.json`.
    ```
    ...
    "workspaces": {
      "packages": [
        "package/"
      ],
    },
    ...
    ```
6. Run `yarn workspaces info` in the root. => Outputs workspace list correctly.
7. Run `yarn workspaces info` in `package/`. => Fails with
    ```
    yarn workspaces v1.6.0
    error Cannot find the root of your workspace - are you sure you're currently in a workspace?
    info Visit https://yarnpkg.com/en/docs/cli/workspaces for documentation about this command.
    ```

**What is the expected behavior?**
Running `yarn workspaces info` should output the workspaces when run both inside root and `package/`.

If this behavior is intended, the documentation could reflect the reason why.

**Please mention your node.js, yarn and operating system version.**
Node v8.11.1
Yarn v1.6.0
Ubuntu 16.04 LTS

**Comment by arcanis** [comment]
Nope, you're right, this is a bug. Do you want to give it a shot at
figuring out what happens and fixing it?

**Comment by Y0hy0h** [comment]
@arcanis I'm not sure if I'll have time, but in principle I'd like to try.

I'll write a new comment before I start, so that you know whether I'm working on this. Until then, if anyone would like to tackle this, go ahead.

**Comment by Y0hy0h** [comment]
I've started looking into it and I assume the root cause lies in [micromatch](https://github.com/yarnpkg/yarn/blob/master/src/config.js#L636).

I will continue diggin for a bit.

**Comment by Y0hy0h** [comment]
This behaviour is [reportedly supported in a newer version of micromatch](https://github.com/micromatch/micromatch/issues/128#issuecomment-388967459).  
The newest version is [3.1.10](https://www.npmjs.com/package/micromatch), whereas yarn is dependent on [2.3.11](https://github.com/yarnpkg/yarn/blob/master/package.json#L30).

I'm not sure how to add a test case for this bug. If you have any suggestions, I could try.

**Comment by Y0hy0h** [comment]
This is [considered intended behavior micromatch](https://github.com/micromatch/micromatch/issues/128#issuecomment-389018254). It thus should be handled in the [call to the function](https://github.com/yarnpkg/yarn/blob/master/src/config.js#L636).

**Comment by 07Gond** [comment]
Hi, @arcanis & @BYK  is this issue still relevant?

**Comment by thomassuckow** [comment]
Just ran into this in v1.10.1

**Comment by joefiorini** [comment]
I also just ran into this issue in yarn v1.15.2. Crazy.

**Comment by forivall** [comment]
I ran into what is probably the same issue, where instead of a trailing `/`, it was caused by a preceding `./`

For example:
```
...
"workspaces": {
  "packages": [
    "./packages/*"
  ],
},
...
```

**Comment by simkessy** [comment]
Currently having this issue with this config: 
```
    "workspaces": [
        "libs/common/client/",
        "services/app/client/"
    ]
```
At the root, I can find the packages.
But at the package level, it doesn't recognize it's part of a workspace. How do I fix this?

If I go to the package and add a dependency, it will create it in the packages node_modules instead of root.

**Comment by forivall** [comment]
@simkessy for now, the workaround is to change that to
```
    "workspaces": [
        "libs/common/client",
        "services/app/client"
    ]
```

**Comment by simkessy** [comment]
Wow, that was simple. This should be documented somewhere, wasted a lot of hours :( 

Thanks @forivall !
